### PR TITLE
Make player face mouse in every state

### PR DESCRIPTION
**Pull request by kepler471**
TIN

... also small cleanup to player.gd, moved some logic out of _physics_process

### CODE REVIEW

File: Moonshot/player/player.gd [hunk 1]
```diff
@@ -86,6 +76,39 @@ func _physics_process(_delta) -> void:
...
+	var mouse_side := get_global_mouse_position().x - get_global_position().x
+	if is_zero_approx(mouse_side):
+		return
+	elif sign(mouse_side) == sign(facing) and playing_reverse:
+		$AnimatedSprite.play($AnimatedSprite.name, false)
+		playing_reverse = false
+	elif sign(mouse_side) == -1 * sign(facing):
+		flip_facing()
+		$AnimatedSprite.play($AnimatedSprite.name, true)
+		playing_reverse = true
```
Comment (kepler471) [inline Moonshot/player/player.gd:109]: This is the new code. Called from line 58.

File: Moonshot/player/player.gd [hunk 2]
```diff
@@ -86,6 +74,39 @@ func _physics_process(_delta) -> void:
...
```
Comment (kepler471) [inline Moonshot/player/player.gd:107]: This is the new code. Called from line 58.

File: Moonshot/player/player.gd [hunk 3]
```diff
@@ -86,6 +74,39 @@ func _physics_process(_delta) -> void:
...
+
+func set_animation() -> void:
+	var new_state = state_machine.get_animation_name()
+	if new_state != null and new_state != animation_name:
+		animation_name = new_state
+		$AnimatedSprite.play(animation_name)
+
+
+func face_mouse() -> void:
+	var mouse_side := get_global_mouse_position().x - get_global_position().x
```
Comment (wattsit) [inline Moonshot/player/player.gd:98]: You can replace `get_global_position().x` with  just `global_position.x` if you want.

**Review by wattsit — APPROVED** [review]
LGTM

**Review by whitew1994WW — APPROVED** [review]

**Pull request merged**